### DISCUSSION TITLE
Compatibility with v4

### DISCUSSION THREAD

**Discussion by caomicc** [VirtusLab/strapi-plugin-navigation#120] [Ideas]
Hi!

Are there any plans to make this plugin work with v4? It currently does not work for me and I'd love to be able to use it eventually :)

**Comment by cyp3rius** [comment]
Yep we're working on v4 re-implementation :)

**Reply by IvanBernatovic** [reply]
@cyp3rius Nice! I took a quick glance at v3 -> v4 migration guide for plugins, it looks relatively straightforward for simple plugins. How hard is going to be to migrate for you?

**Reply by cyp3rius** [reply]
Because of poly relations and plugin complexity it's not so straightforward for us :) We've decided to re-write whole plugin logic to fit v4 requirements because of it.

**Reply by IvanBernatovic** [reply]
Your plugin is definitely not a simple one (it solves a relatively complex problem) so it's understandable. I asked because I wanted to check if it makes sense to start a new v4 project and wait for this plugin (must-have for my project) or simply start v3 and use your plugin from the get-go. Seems like it could take a few months so maybe sticking with v3 is my best bet.

**Reply by cyp3rius** [reply]
We would like to drop first beta between Christmas and New Year :) Working hard with my team to manage it. Promise is a promise :D

**Reply by IvanBernatovic** [reply]
Awesome! But don't forget to take some rest for the holidays... No one is really gonna do anything major with their projects in the next few weeks 😄

**Comment by cyp3rius** [comment]
Folks, we've just released `v2.0.0-beta.1` with v4 support! Please read the [release notes](https://github.com/VirtusLab-Open-Source/strapi-plugin-navigation/releases/tag/v2.0.0-beta.1).

**Comment by cyp3rius** [comment]
Closing the discussion as we already got it 💪
